### ISSUE TITLE
Web Uploaded Photos Fail To Enter SQL Database at piwigo_image_category, but FTP Upload and Synchronization Works

### ISSUE THREAD

**Issue by hereiam-at-mit-dot-edu** [Piwigo/Piwigo#2154]
I've been chasing an issue which appears persistent from PiwiGo version 13.x through the current 14.4

When I upload photos through the web interface, I can see that all steps but one seem to complete:

1. Photos make it into the uploads/ directory with correct permissions
2. Thumbnails are generated in the _data folder correctly

However, two SQL database updates seem not to occur:

1. In piwigo_images, the field storage_category_id remains NULL rather than being set to the appropriate album (category).
2. In piwigo_image_category, no new entry is created.

When I manually fix the storage_category_id in piwigo_images, AND manually create the correct entry linking the image_id and category_id fields in the piwigo_image_category table, everything works as it should.

Finally, bulk uploaing photos via FTP and manually synchronizing through the Administration interface all works perfectly. The photos uploaded through the web interface are not picked up by this synchronization.

Would love to know what the issue might be, or what I may try to patch and submit this! Or I can start going through the code myself, though if a friendly human index out there could point me towards some of the right files that would be a big timesaver. Thank you!!!

**Comment by hereiam-at-mit-dot-edu** [comment]
Interesting, do you mean adding photos with the batch manager (without pre-selecting the album from the album manager) works, but clicking the Edit button from album manager and then trying to upload doesn't? That could help narrow down the source of the issue. Thanks!

**Comment by hampoelz** [comment]
Exactly, if I upload the photos directly to an album, the photos are not added to the album. But if I then use the batch manager to "filter for photos without a virtual album" and the "move to album" action, they are assigned to the album.

I also tried removing all plugins as well as completely reinstalling the Piwigo instance, except for the database (and the galleries folder), but that didn't change everything. However, reinstalling with resetting the database “fixed” the issue.

So I thought that the database must be corrupted somehow and ran the built-in database repair & optimization, integrity check, and the "RV DB Integrity" plugin, but none of those detected a problem.
